pkg/cluster/cluster.go

```diff
@@ -281,5 +287,110 @@ func New(ctx context.Context, log *logrus.Entry, _env env.Interface, db database
 		hiveClusterManager:                hiveClusterManager,
 		now:                               func() time.Time { return time.Now() },
 		openShiftClusterDocumentVersioner: new(openShiftClusterDocumentVersionerService),
-	}, nil
+	}
+
+	if doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		cloud := ""
+		switch strings.ToUpper(_env.Environment().Name) {
+		case dataplane.AzurePublicCloud:
+			cloud = dataplane.AzurePublicCloud
+		case dataplane.AzureUSGovCloud:
+			cloud = dataplane.AzureUSGovCloud
+		}
+
+		if cloud == "" {
+			return nil, errors.New("could not determine which Azure Cloud to use to instantiate MSI dataplane client")
+		}
+
+		msiDataplaneClientOptions := clientOptions.ClientOptions
+
+		if _env.FeatureIsSet(env.FeatureUseMockMsiRp) {
```

Comment:
Can we also move the validatevars to here:- https://github.com/Azure/ARO-RP/blob/master/cmd/aro/rp.go#L73
So that we validate these vars when RP comes up in dev rather than for every cluster creation?

Reply:
>Can we also move the validatevars to here:- https://github.com/Azure/ARO-RP/blob/master/cmd/aro/rp.go#L73
So that we validate these vars when RP comes up in dev rather than for every cluster creation?

I'd rather leave it where it is because if we validate the vars on RP startup we'd be forcing people who don't need to create MIWI clusters to set env vars they aren't going to use. Since setting the env vars correctly is non-trivial, I find it easy to imagine that people would end up setting them to nonsensical values just to get the pre-startup validation to work, which would be an annoying extra step to start a local dev or INT RP.

Reply:
>I would make a new method in env that returns the mocked ClientOptions, and use it here.

I'll still be making this change though.

Reply:
Done with the `ClientOptions` change 👍🏻